virttest/qemu_devices/qcontainer.py

```diff
@@ -2548,9 +2549,18 @@ def dimm_device_define_by_params(self, params, name):
         params = params.object_params("dimm")
         dimm_type = "nvdimm" if params.get("nv_backend") else "pc-dimm"
         attrs = qdevices.Dimm.__attributes__[:]
-        params = params.copy_from_keys(attrs)
-        dev = qdevices.Dimm(params=params, dimm_type=dimm_type)
+        dev = qdevices.Dimm(params=params.copy_from_keys(attrs),
+                            dimm_type=dimm_type)
         dev.set_param("id", "%s-%s" % ("dimm", name))
+        if dimm_type == "nvdimm" and params.get("nvdimm_uuid"):
```

Comment:
Since the option is supported by qemu-5.0 or later, shall we add a check to judge whether should express it in command line or not?

Reply:
I don't think so, although this option is supported since QEMU 5.0, it's not a required one except for pseries, so I set `nvdimm_uuid = ""` in base.cfg. If users want to set the UUID for nvdimm device, they can modify their configure file by themselves in testcase side.

Reply:
Here is an issue, by setting this param in `machines.cfg` for pseries guests, users run tests with qemu less than 5.0 would get the failure since the option is not supported, isn't it?

Reply:
This case will not happen.
pseries supports nvdimm device beginning at [qemu-5.0](https://wiki.qemu.org/ChangeLog/5.0#PowerPC), which means we need to add a filter `required_qemu = [5.0, ]` in test case cfg. If users want to launch a guest with nvdimm device less than qemu-5.0, qemu will be terminated and report that there is not this device.

**QEMU 4.2:**
```bash
$ /usr/libexec/qemu-kvm -device ? | tail -n 3
Uncategorized devices:
name "pc-dimm", desc "DIMM memory module"
name "spapr-tpm-proxy"
```

**QEMU 5.0:**
```bash
$ /usr/libexec/qemu-kvm -device ? | tail -n 4
Uncategorized devices:
name "nvdimm", desc "DIMM memory module"
name "pc-dimm", desc "DIMM memory module"
name "spapr-tpm-proxy"
```

Reply:
ok, thanks for your explanation!